pyccel/ast/core.py

```diff
@@ -3743,6 +3745,72 @@ def __str__(self):
                 args   = ', '.join(self.args),
                 result = result)
 
+class FunctionAddress(FunctionDef):
+    def __init__(
+        self,
+        name,
+        arguments,
+        results,
+        body,
+        is_optional=False,
+        is_pointer=False,
+        is_kwonly=False,
+        is_argument=False,
+        *args,
+        **kwargs
```

Comment:
It seems odd to add `*args` here this means that you have to specify `is_optional`, `is_pointer`, `is_kwonly`, and `is_argument` in order to provide additional args. I think it would make more sense to have these 4 as keyword only arguments so:
```
def __init__(
        self,
        name,
        arguments,
        results,
        body,
        *args,
        is_optional=False,
        is_pointer=False,
        is_kwonly=False,
        is_argument=False,
        **kwargs)
```